lib/charms/opensearch/v0/models.py

```diff
@@ -40,23 +87,118 @@ def is_data(self):
+    """Nature of a sub cluster deployment."""
 
-    @staticmethod
-    def from_dict(input_dict):
-        """Create a new instance of this class from a json/dict repr."""
-        return Node(input_dict.get("name"), input_dict.get("roles"), input_dict.get("ip"))
+    MAIN_CLUSTER_MANAGER = "main-cluster-manager"
+    CLUSTER_MANAGER_FAILOVER = "cluster-manager-failover"
+    OTHER = "other"
 
-    @staticmethod
-    def from_str(input_str_dict):
-        """Create a new instance of this class from a stringified json/dict repr."""
-        return Node.from_dict(json.loads(input_str_dict))
+
+class StartMode(BaseStrEnum):
+    """Mode of start of units in this deployment."""
+
+    WITH_PROVIDED_ROLES = "start-with-provided-roles"
+    WITH_GENERATED_ROLES = "start-with-generated-roles"
+
+
+class Directive(BaseStrEnum):
+    """Directive indicating what the pending actions for the current deployments are."""
+
+    NONE = "none"
+    SHOW_STATUS = "show-status"
+    WAIT_FOR_PEER_CLUSTER_RELATION = "wait-for-peer-cluster-relation"
+    INHERIT_CLUSTER_NAME = "inherit-name"
+    VALIDATE_CLUSTER_NAME = "validate-cluster-name"
+    RECONFIGURE = "reconfigure-cluster"
+
+
+class State(BaseStrEnum):
+    """State of a deployment, directly mapping to the juju statuses."""
+
+    ACTIVE = "active"
+    BLOCKED_WAITING_FOR_RELATION = "blocked-waiting-for-peer-cluster-relation"
+    BLOCKED_WRONG_RELATED_CLUSTER = "blocked-wrong-related-cluster"
+    BLOCKED_CANNOT_START_WITH_ROLES = "blocked-cannot-start-with-current-set-roles"
+    BLOCKED_CANNOT_APPLY_NEW_ROLES = "blocked-cannot-apply-new-roles"
+
+
+class DeploymentState(Model):
+    """Full state of a deployment, along with the juju status."""
+
+    value: State
+    message: str = Field(default="")
+
+    @root_validator
+    def prevent_none(cls, values):  # noqa: N805
+        """Validate the message or lack of depending on the state."""
+        if values["value"] == State.ACTIVE:
+            values["message"] = ""
+        elif not values["message"].strip():
+            raise ValueError("The message must be set when state not Active.")
+
+        return values
+
+
+class PeerClusterRelDataCredentials(Model):
+    """Model class for credentials passed on the PCluster relation."""
+
+    admin_username: str
+    admin_password: str
+
+
+class PeerClusterRelData(Model):
+    """Model class for the PCluster relation data."""
+
+    cluster_name: Optional[str]
+    cm_nodes: List[str]
+    credentials: PeerClusterRelDataCredentials
+    tls_ca: str
+
+
+class PeerClusterConfig(Model):
+    """Model class for the multi-clusters related config set by the user."""
+
+    cluster_name: str
+    init_hold: bool
+    roles: List[str]
+    data_temperature: Optional[str] = None
+
+    @root_validator
+    def set_node_temperature(cls, values):  # noqa: N805
+        """Set and validate the node temperature."""
+        allowed_temps = ["hot", "warm", "cold", "frozen"]
```

Comment:
I'm not sure how to do that with pydantic, it seems we can have it as a `Field(.., const=True)` but then this would be (de)serialized.. Otherwise a plain `@property` but then it probably wouldn't be worth having it.. 
Thoughts?

Reply:
Well, I just meant that it is a bit better to do the following:

```
class TestClass(BaseModel):
     _VALUES=["a", "b"]
     a: str
     @root_validator
     def _validate(cls, values):
         if values["a"] not in cls._VALUES:
             raise TypeError()
         return values
```

Or something along this line